packages/thirdweb/src/extensions/lens/read/getFullProfile.ts

```diff
@@ -0,0 +1,90 @@
+import type { ThirdwebClient } from "../../../client/client.js";
+// import { getDefaultHandle } from "src/exports/extensions/lens.js";
+import type { ThirdwebContract } from "../../../contract/contract.js";
+import { getProfile } from "../__generated__/LensHub/read/getProfile.js";
+import { getDefaultHandle } from "../__generated__/TokenHandleRegistry/read/getDefaultHandle.js";
+import type { LensProfileSchema } from "./type.js";
+
+export type GetFullProfileParams = {
+  profileId: bigint;
+  client: ThirdwebClient;
+
+  // Lens's smart contracts.
+  // Go here for the latest data: https://www.lens.xyz/docs/resources/smart-contracts
+  lensHubContract: ThirdwebContract;
+  lensHandleContract: ThirdwebContract;
+  tokenHandleRegistryContract: ThirdwebContract;
+
+  includeJoinDate?: boolean;
+  // includeFollowerCount?: boolean;
+  // includeFollowingCount?: boolean;
+};
+
+export type FullProfileResponse =
+  | (LensProfileSchema & {
+      handle: string;
+      // Timestamp of the join date
+      joinDate: bigint | null;
+    })
+  | null;
+
+/**
+ * Return the profile data _with Lens handle_ and optional join date + follower/following count
+ *
+ * In Lens Protocol, which profile is associated with an ERC721 token,
+ * thus, the tokenId represent profileId and the 2 can be used interchangeably
+ * @extension LENS
+ *
+ * @example
+ * ```ts
+ * import { getFullProfile } from "thirdweb/extension/lens";
+ *
+ * const lensProfile = await getFullProfile({ ... });
+ * ```
+ */
+export async function getFullProfile(
+  options: GetFullProfileParams,
+): Promise<FullProfileResponse> {
+  const {
+    profileId,
```

Comment:
how do i get the profileId of an address?

Reply:
The profileId is the tokenId of the LensHub contract (is an erc721)
Tbh there's no easy way to query this onchain - we have to resort to an indexing solution which is what lens SDK does